File: dotnet/windntrees.core/Application.Core/wwwroot/js/models/Role.js
function Role(data) {
    var instance = this;
    instance.Id = ko.observable(data.Id);
    instance.Name = ko.observable((data.Name !== null && data.Name !== undefined) ? data.Name : data.Id);
    instance.NormalizedName = ko.observable(data.NormalizedName);
    instance.ConcurrencyStamp = data.ConcurrencyStamp;
}

File: dotnet/windntrees.core/Application.Core/Models/Role.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using SharedLibrary.Core.Attributes;

namespace Application.Core.Models
{
    public class Role
    {
        [LocaleMessageRequired]
        public string RoleId { get; set; }
        [LocaleMessageRequired]
        public string Name { get; set; }

        public string ConcurrencyStamp { get; set; }
        public string NormalizedName { get; set; }
    }
}